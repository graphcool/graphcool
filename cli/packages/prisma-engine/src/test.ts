import { Engine } from './Engine'

async function run() {
  const engine = new Engine()
  console.log('Starting Engine')
  await engine.start()
  console.log('Engine started')
  const result = await engine.request(
    `
    {
      cats {
        id
      }
    }
  `,
  )

  console.log({ result })
  engine.stop()
}

run().catch(e => console.error(e))
