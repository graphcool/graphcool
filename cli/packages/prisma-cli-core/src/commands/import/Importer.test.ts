import * as path from 'path'
import { Importer } from './Importer'
import { Environment } from 'prisma-yml'
import { Client, Config, Output } from 'prisma-cli-engine'
import { getTmpDir } from '../../../../prisma-yml/src/test/getTmpDir'

function initImporter() {
  const config = new Config()
  const output = new Output(config)
  const env = new Environment(getTmpDir(), output)
  const client = new Client(config, env, output)
  const types = `
    type Post {
      title: String
      description: String
      state: String
    }
  `
  return new Importer(
    path.join(__dirname, 'fixtures', 'basic', 'import-missing-ids'),
    types,
    client,
    output,
    config,
  )
}

describe('Importer', () => {
  it('Should throw an error when ids are missing', async () => {
    const importer = initImporter()
    await importer.upload('serviceName', 'stage')
    expect(importer.out.stdout.output).toMatchSnapshot()
  })

  it('Should pass the validatation and start uploading nodes', async () => {
    const importer = initImporter()
    await importer.upload('serviceName', 'stage', undefined, undefined, true)
    expect(importer.out.stdout.output).toContain('Uploading nodes...')
  })
})
