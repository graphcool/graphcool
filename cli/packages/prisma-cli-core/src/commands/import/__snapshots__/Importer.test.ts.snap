// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Importer Should throw an error when ids are missing 1`] = `
"Uncaught exception, cleaning up: Error: Object {\\"_typeName\\":\\"Post\\",\\"title\\":\\"Quas dolores earum corporis aut voluptatibus dicta voluptatem.\\",\\"description\\":\\"Nulla non esse vitae hic sunt ut eveniet rem eaque. Et qui cumque tempora dolorem nisi et distinctio dolores est. Quia est commodi omnis voluptatem aspernatur voluptatem blanditiis perspiciatis. Qui qui earum consequatur illo molestias numquam dolores voluptas reprehenderit. Perferendis ipsum deserunt dolorem et. Et deleniti facere nesciunt reprehenderit accusamus.\\\\n \\\\rVero repudiandae blanditiis. Ab aut distinctio ut qui rem. Aut accusantium sed aliquid sit amet vero. Quasi impedit eum.\\\\n \\\\rItaque quas cupiditate animi amet non recusandae voluptatem non facere. Omnis quo eaque autem labore laborum qui incidunt. Et minima ut veritatis rerum aut architecto sint non accusamus.\\",\\"state\\":\\"Published\\"} needs an id property
"
`;
